src/Components/MainHeader/MiddleMenu/SearchBar/SearchBar.js

```diff
@@ -1,73 +1,116 @@
 import './SearchBar.css';
-import React from 'react';
-// 숙소 버튼 클릭 시
+import Box from '../Function/FlexibleSearch/Flexible';
+import Add from '../Function/NOP/Nop';
+import CalenderModal from '../Function/Calender/Calender';
+import { useEffect, useRef, useState } from 'react';
+
 export const Lodge = () => {
+  const el = useRef();
+  const [search, setSearch] = useState(false);
+  const onToggle = () => {
+    setSearch((search) => !search);
+  };
+  const [add, setAdd] = useState(false);
+  const onButton = () => {
+    setAdd((add) => !add);
+  };
+  const [calender, setCalender] = useState(false);
+  const onCalender = () => {
+    setCalender((calender) => !calender);
+  };
+  const closedModal = (e) => {
+    if (el.current && !el.current.contains(e.target)) {
+      setSearch(false);
+      setAdd(false);
+      setCalender(false);
+    }
+  };
+
+  useEffect(() => {
+    window.addEventListener('mousedown', closedModal);
```

Comment:
보통은 window에 이벤트리스너를 추가하지 않고 document에 추가합니다
```suggestion
    document.addEventListener('mousedown', closedModal);
```
가끔 window 객체에 이벤트 리스너를 달아야 하는 경우가 있는데요, 현재 상황에서는 모르셔도 괜찮을 것 같습니다.